### PR TITLE
Migrate codegen to pure-rust generator

### LINKED ISSUE #2: replace with quick-protobuf
pure Rust & faster, means we can use it inside WASM / port it more easily https://github.com/tafia/quick-protobuf

### PR DESCRIPTION
**Pull request by bltavares**
This commit replaces the build.rs dev dependency to generate protobuf
files using a pure-rust implementation, removing the requirement to have
`protoc` installed and making it easier to contribute.

This commit also bumps the lib protobuf lib to the latest version, and pin it
to a specific dependency, as required by the constant it generates. Due
to requiring a specific protobuf version, a new minor version has been
bumped.

Tentatively fix https://github.com/datrs/dat-network-protocol/issues/2
---
**Choose one:**  a 🙋 feature

## Checklist

- [x] tests pass
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added

## Context
https://github.com/datrs/dat-network-protocol/issues/2

## Semver Changes
`0.2.0`

### CODE REVIEW

File: src/cancel.rs [hunk 1]
```diff
@@ -180,13 +191,13 @@ impl ::protobuf::Message for Cancel {
...
+    fn as_any(&self) -> &dyn (::std::any::Any) {
+        self as &dyn (::std::any::Any)
     }
-    fn as_any_mut(&mut self) -> &mut ::std::any::Any {
-        self as &mut ::std::any::Any
+    fn as_any_mut(&mut self) -> &mut dyn (::std::any::Any) {
+        self as &mut dyn (::std::any::Any)
     }
-    fn into_any(self: Box<Self>) -> ::std::boxed::Box<::std::any::Any> {
+    fn into_any(self: Box<Self>) -> ::std::boxed::Box<dyn (::std::any::Any)> {
```
Comment (yoshuawuyts) [inline src/cancel.rs:200]: Did you run `cargo fmt` on the files? With `edition=2018` the `::prefix` on paths should have been removed.

**Review by yoshuawuyts — APPROVED** [review]
Looks good otherwise!

Do you have merge / publish access? Can provide if you don't already (:

File: src/cancel.rs (hunk 1 above)
Comment (bltavares) [reply src/cancel.rs:200]: I haven't run anything, just the generation steps from the lib. I could take a look on that later on and try to fix the CI as well

**Comment by bltavares** [comment]
I don't have access here, and I'm not sure how to publish crates either, but I could learn :)

**Comment by yoshuawuyts** [comment]
@bltavares I've given you access to both! If you log into crates.io you're now part of the team for  https://crates.io/crates/dat-network-protocol :tada:

**Pull request merged**